### DISCUSSION TITLE
Does coil handle request coalescing?

### DISCUSSION THREAD

**Discussion by numsu** [coil-kt/coil#2533] [General]
As per the title, does it? It is not mentioned in the docs or anywhere online.

For example, when requesting the same image multiple times from the same client at the same time and the image is not found in a cache, coil should only send a single network request, but dispatch the same response to all client requests.

**Comment by colinrtwhite** [comment]
Coil doesn't pool network requests to the same image. There's some edge cases that make it slightly complex to support this. For instance, if a request disables disk caching we need to share the response body, however a network response body can only be read once (caching it in memory could OOM for large images). Also, how do we decide which URLs to pool? Do we pool URLs to the same resource, but with different headers? I don't believe Glide or Picasso support this either for similar reasons.
